internal/browser/browser.go

```diff
@@ -26,11 +28,40 @@ type Opener interface {
 
 type StderrPrint struct{}
 
+func openURL(url string) error {
+	switch runtime.GOOS {
+	case "darwin": // macOS
+		return exec.Command("open", url).Run()
+	case "linux":
+		providers := []string{"xdg-open", "x-www-browser", "www-browser"}
+
+		// There are multiple possible providers to open a browser on linux
+		// One of them is xdg-open, another is x-www-browser, then there's www-browser, etc.
+		// Look for one that exists and run it
+		for _, provider := range providers {
+			if binPath, err := exec.LookPath(provider); err == nil {
+				err = exec.Command(binPath, url).Run()
+				if err == nil {
+					return nil
+				}
+			}
+		}
+		return fmt.Errorf("unsupported platform")
+	case "windows":
+		return exec.Command("rundll32", "url.dll,FileProtocolHandler", url).Run()
+	default:
+		return fmt.Errorf("unsupported platform")
+	}
+}
+
 func (p *StderrPrint) Open(u *url.URL) error {
 	fmt.Fprintf(
 		os.Stderr,
-		"Please open the following URL in your web browser to authenticate:\n\n\t%s\n\n",
+		`Attempting to automaticaly open the authentication URL in your web browser.
+If the browser does not open or you wish to use a different device to authorize this request, open the following URL:
+		%s`,
```

Comment:
Two tabs ends up being a lot of space. Let's take it back to one.

Also make sure to keep the newline at the end here.